# HG changeset patch
# User Torsten <ttl@justmail.de>
# Date 1451301005 -3600
#      Mon Dec 28 12:10:05 2015 +0100
# Node ID dd1dcef1ee335b52afba081179b745a3a662ef97
# Parent  31a12af0aaafd649acbc523d8dbbeda6b07f97aa
prevent terminal context menu from exceeding the window

* QTerminal.h: use exec () instead of show () for showing the context menu

diff --git a/libgui/qterminal/libqterminal/QTerminal.h b/libgui/qterminal/libqterminal/QTerminal.h
--- a/libgui/qterminal/libqterminal/QTerminal.h
+++ b/libgui/qterminal/libqterminal/QTerminal.h
@@ -104,18 +104,17 @@ public slots:
 
   virtual void handleCustomContextMenuRequested (const QPoint& at)
   {
     QClipboard * cb = QApplication::clipboard ();
 
     _paste_action->setEnabled (cb->text().length() > 0);
     _copy_action->setEnabled (selectedText().length() > 0);
 
-    _contextMenu->move (mapToGlobal (at));
-    _contextMenu->show ();
+    _contextMenu->exec (mapToGlobal (at));
   }
 
   void notice_settings (const QSettings *settings);
 
   virtual void init_terminal_size (void) { }
 
   void terminal_interrupt (void) { emit interrupt_signal (); }
 
